src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+    { { 255, 0, 0 }, object_link_goron_DL_011AB8 },
+};
+#endif
+
+void func_80122D44(GlobalContext* globalCtx, struct_80122D44_arg1* arg1) {
+    struct_801BFDD0* temp_s3;
+    struct_80122D44_arg1_unk_04* phi_s2 = arg1->unk_04;
+    s32 phi_s6 = false;
+    s32 i;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    for (i = 0; i != ARRAY_COUNT(arg1->unk_04); i++) {
+        if ((phi_s2->unk_01 != 0) && (phi_s2->unk_01 != 0xFF)) {
+            temp_s3 = &D_801BFDD0[phi_s2->unk_00 - 1];
+            Matrix_SetCurrentState(&phi_s2->unk_04);
+
+            gDPPipeSync(POLY_XLU_DISP++);
+
+            if (!phi_s6 && phi_s2->unk_00 == 2) {
+                AnimatedMat_DrawXlu(globalCtx, Lib_SegmentedToVirtual(&object_link_goron_Matanimheader_013138));
+                phi_s6 = true;
+            }
+
+            Scene_SetRenderModeXlu(globalCtx, 1, 2);
+            gDPSetEnvColor(POLY_XLU_DISP++, temp_s3->color.r, temp_s3->color.g, temp_s3->color.b, phi_s2->unk_01);
+
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+            gSPDisplayList(POLY_XLU_DISP++, temp_s3->dlist);
+        }
+
+        phi_s2++;
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+extern u8 D_801BFDE8[PLAYER_MASK_MAX - 1];
+#if 0
+u8 D_801BFDE8[PLAYER_MASK_MAX - 1] = {
+    ITEM_MASK_TRUTH,         // PLAYER_MASK_TRUTH
+    ITEM_MASK_KAFEIS_MASK,   // PLAYER_MASK_KAFEIS_MASK
+    ITEM_MASK_ALL_NIGHT,     // PLAYER_MASK_ALL_NIGHT
+    ITEM_MASK_BUNNY,         // PLAYER_MASK_BUNNY
+    ITEM_MASK_KEATON,        // PLAYER_MASK_KEATON
+    ITEM_MASK_GARO,          // PLAYER_MASK_GARO
+    ITEM_MASK_ROMANI,        // PLAYER_MASK_ROMANI
+    ITEM_MASK_CIRCUS_LEADER, // PLAYER_MASK_CIRCUS_LEADER
+    ITEM_MASK_POSTMAN,       // PLAYER_MASK_POSTMAN
+    ITEM_MASK_COUPLE,        // PLAYER_MASK_COUPLE
+    ITEM_MASK_GREAT_FAIRY,   // PLAYER_MASK_GREAT_FAIRY
+    ITEM_MASK_GIBDO,         // PLAYER_MASK_GIBDO
+    ITEM_MASK_DON_GERO,      // PLAYER_MASK_DON_GERO
+    ITEM_MASK_KAMARO,        // PLAYER_MASK_KAMARO
+    ITEM_MASK_CAPTAIN,       // PLAYER_MASK_CAPTAIN
+    ITEM_MASK_STONE,         // PLAYER_MASK_STONE
+    ITEM_MASK_BREMEN,        // PLAYER_MASK_BREMEN
+    ITEM_MASK_BLAST,         // PLAYER_MASK_BLAST
+    ITEM_MASK_SCENTS,        // PLAYER_MASK_SCENTS
+    ITEM_MASK_GIANT,         // PLAYER_MASK_GIANT
+    ITEM_MASK_FIERCE_DEITY,  // PLAYER_MASK_FIERCE_DEITY
+    ITEM_MASK_GORON,         // PLAYER_MASK_GORON
+    ITEM_MASK_ZORA,          // PLAYER_MASK_ZORA
+    ITEM_MASK_DEKU,          // PLAYER_MASK_DEKU
+};
+#endif
+
+u8 func_80122ED8(s32 maskIdMinusOne) {
+    return D_801BFDE8[maskIdMinusOne];
+}
+
+u8 Player_CurMaskToItemId(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (player->currentMask != PLAYER_MASK_NONE) {
+        return func_80122ED8(player->currentMask - 1);
+    }
+
+    return ITEM_NONE;
+}
```

Comment:
Seems like the first of these is MaskIdToItemId, and the second is something like GetCurMaskItemId. Also, do we want the prefix for this stuff to be PlayerLib? I think it would be helpful to disambiguate it, even though we know that almost nothing in Player can be accessed from outside it.

Reply:
I think we would need to discuss if we want to use PlayerLib as a prefix for functions in this file